src/components/DatasetCounter.tsx

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { datasetCount } from "@/services/ckan/index.server";
+
+const DatasetCounter = () => {
+  const [count, setCount] = useState<number | null>(null);
```

Comment:
You don't need a state if you make it a server side component 

You can just do

const count = await datasetCount(); and then directly use it in component body